controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -2736,9 +2736,51 @@ func TestHFSTransitionToPreparing(t *testing.T) {
 			{Type: "ChangeDetected", Status: "True", Reason: "Success"},
 			{Type: "Valid", Status: "True", Reason: "Success"},
 		},
+		Settings: metal3v1alpha1.SettingsMap{
+			"ProcVirtualization": "Enabled",
+			"SecureBoot":         "Enabled",
+		},
 	}
 
 	r.Update(goctx.TODO(), hfs)
 
 	waitForProvisioningState(t, r, host, metal3v1alpha1.StatePreparing)
 }
+
+// TestHFSEmptyStatusSettings ensures that BMH does not move to the next state in the
+// following two senarios:
+// 1. when a user provides the BIOS settings on a hardware server that does not
+// have the required license to configure BIOS
```

Comment:
This is really the same scenario, its just that when for example the license is not provided for certain hardware there will never be data coming back from Ironic as opposed to just a delay waiting for Ironic to retrieve the data from the BMC.

Reply:
The comment in TestHFSEmptyStatusSettings has been updated.